### DISCUSSION TITLE
最快 IP 同一个运营商不同地区，差距有多大？适用范围多广？出境路由是否相同？

### DISCUSSION THREAD

**Discussion by CrazyBoyFeng** [XIU2/CloudflareSpeedTest#52] [讨论]
北京移动测到的最好ip，是不是对天津移动、上海移动、广东移动也是最好的？
如果不同地区的出境路由是不同的，那么这个粒度是多大？几个省一个大区？还是每个省都不一样？

我知道商宽和家宽的出境路由是不同的。不过我们这里寻找最好的ip，一般是针对家宽而言的。商宽往往有针对出国路由的特别优化，找ip的需求并不是非常紧要。
所以我们只讨论全国各地同一运营商的家宽出境路由。

**Comment by XIU2** [comment]
这个需要多地配合一起测试才知道。

不断地调整变量来测试。
比如，先找**当地同运营商**的测试，再找**外地同运营商**的测试。

****

另外，你还不能只看路由是否一样，你还要看**当地运营商的 QOS** 情况（即动态拥塞控制，或者说**动态限速**）。

同一个 IP，同一个运营商，同一个出境路由方向（比如都是上海出境），速度差别很大时，往往都是**运营商 QOS** 的结果。

****

至于**当地不同运营商、外地不同运营商**的就比较悬了，有条件也可以测试，没条件就算了。

**Reply by CrazyBoyFeng** [reply]
同一个地区的不同运营商出境路由肯定是不一样的，即便移动联通有可能会走CN2GIA，那也是跟电信购买了BGP，效果好不好取决于买了多少带宽。会存在移动联通走CN2GIA比当地电信家宽还快的情况。

**Reply by XIU2** [reply]
刚刚我在跟一个人远程协助的时候，我们两个都是联通，但是他的测速结果就很捉急。。。
同一个 IP，我测速有 25MB/s，他却只有 1MB/s 左右，都是北方联通。

**Comment by XIU2** [comment]
很多人嫌我写的太长，那我就总结一下（延迟和速度仅代表我个人）：

> ***已失效，该日本 IP 段已被 Cloudflare CDN 官方限制，无法在用于自选 IP（只提供给使用 CDN 付费套餐的网站）。***

||||
-- | -- | --
**IP** | ~`172.64.228.0/24`~ | 整个 IP 段共 **256** 个 IP（0~255），质量都一样
**地区** | 日本东京 | colo=NRT（随时可能**动态调整**为其他地区）
**线路** | NTT | 在各中日线路中算比较**垃圾**的（高峰期拥堵严重）**Vultr 日本**就是这个，但更好点
**丢包率** | 白天 `10%` 晚上 `30%` | **拉低**下载速度（最近发现白天的**丢包率**突然变得**升高**）
**平均延迟** | `±108 ms` | 存在**向上**波动（有时候会突然变成 `150ms+`...）
**平均速度** | `±4.5 MB/s` | 因为始终存在**丢包**，所以速度上不去
**适用范围** | 全国大部分地区 | 比 CDN 默认分配 IP **延迟更低点**，具体数据看最下面 
**总体质量** | 中等偏下 | 除了**延迟低**外没什么优点

****

刚才无意间扫到一个 IP 发现延迟很低（**平均 108ms**，而其他的 IP 最低 200ms+），在这个高峰期阶段鹤立鸡群...

查了下，发现这个 IP 的**整个 IP 段**都是这么低延迟，此时该 IP 段的物理位置是**日本**。

> 因为 Cloudflare CDN 用的是 **Anycast**（任播技术），该 IP 段可能随时就会**动态调整为其他地区**了。

```
172.64.228.0/24
```

> **附查询 CDN IP 当前物理位置方法**：访问 `https://IP/cdn-cgi/trace` 看到 `colo=XXX`，`XXX` 就是当地机场缩写。

****

这是我用 CloudflareST 以来，自己测速得到的最低延迟的 IP，而且还是**晚上高峰期**，以前白天低峰期最低也有 130ms+。

不过可能是因为处于高峰期的原因，这个 IP 段也存在**丢包（30% 左右），延迟波动**的情况（这个时间段没有一个不丢包的）。

> 当然，这个 Cloudflare CDN 的日本 IP 虽然**直连（垃圾 NTT 线路）**，但相比其他的日本服务器，**质量依然处于下游水平**。

****

我用 [IPIP.NET](https://tools.ipip.net/pingpro.php) 对比 PING 测试了一下。
发现在国内，相比 Cloudflare CDN **默认分配 IP**，大部分地区**日本 IP 延迟更低**。

全国各地共 **175** 个服务器同时 PING 得到的结果是：日本 IP 在 **169 个服务器中延迟低于**默认分配 IP。
甚至出现了两边延迟差距有 **128ms+** 之多 ......

> 当然，只是延迟相比更低一些，但是**下载测速的话就一般般**了，距离最快的 IP 还有一定差距。

<details>
<summary><strong><code>「 点击查看 对比表格 」</code></strong></summary>

****

> 为了方便观看，精简了不必要的内容，左边是**日本 IP**，右边是**默认分配 IP**，单位 ms

Ping节点 | IP: 172.64.228.123(平均延迟 122.63) | IP: 104.21.81.221(平均延迟 234.75)
-- | -- | --
北京(华为云) | 71.601(丢包:25) | 199.112
湖南长沙(移动云) | 36.898(丢包:25) | 229.120(丢包:75)
重庆(联通) | 101.976 | 298.118
云南昆明(联通) | 113.072 | 296.605(丢包:50)
内蒙古呼和浩特(移动云) | 69.488 | 213.452(丢包:25)
安徽芜湖(电信) | 130.550(丢包:50) | 245.376(丢包:75)
上海地面通(BGP) | 193.855(丢包:25) | 229.714(丢包:50)
浙江杭州(天翼云) | 102.633(丢包:25) | 214.456(丢包:75)
福建厦门(电信) | 185.494 | 236.190(丢包:25)
四川乐山(移动) | 95.639 | 247.112(丢包:40)
广东广州(青云) | 187.303(丢包:75) | 201.166(丢包:25)
山东青岛(联通) | 158.749(丢包:20) | 323.577(丢包:60)
广东茂名(联通)2 | 182.031(丢包:25) | 257.193(丢包:25)
广西南宁(联通) | 99.692 | 310.365(丢包:25)
湖北襄阳(电信) | 196.308(丢包:50) | 226.936(丢包:75)
江苏徐州(电信) | 228.972(丢包:25) | 259.659
吉林延边(联通) | 122.153(丢包:20) | 369.970(丢包:25)
北京(bgp) | 56.266(丢包:50) | 326.176
广东深圳(天威-bgp) |   |  
江西南昌(炎黄盛世) | 209.075 | 240.335(丢包:50)
江苏扬州(联通) |   |  
广东广州(腾讯云-轻量) | 173.843(丢包:25) | 209.233
江苏宿迁(电信) | 83.273 | 271.974(丢包:25)
山东青岛(天翼云) | 136.357(丢包:50) | 251.867(丢包:25)
广东深圳(电信) |   | 223.599(丢包:75)
浙江杭州(阿里云) | 50.666(丢包:50) |  
吉林长春(移动云) | 59.388 |  
北京(腾讯云-轻量) | 70.011(丢包:25) | 200.187(丢包:50)
江苏南京(腾讯云) | 108.220(丢包:25) | 225.264(丢包:75)
浙江宁波(移动云) | 32.807(丢包:25) |  
甘肃兰州(天翼云) | 86.636 | 184.964(丢包:25)
广东广州(华为云) | 235.945(丢包:50) | 208.964(丢包:25)
海南海口(天翼云) | 208.794(丢包:50) | 227.185(丢包:25)
安徽芜湖(天翼云) | 128.788(丢包:25) | 244.670(丢包:50)
四川德阳(移动) | 58.586 | 218.412(丢包:50)
河南洛阳(bgp) | 78.408(丢包:25) | 319.535(丢包:25)
浙江嘉兴(电信) | 110.129(丢包:25) | 228.974(丢包:25)
广东深圳(bgp) | 116.930 | 275.174(丢包:25)
四川成都(腾讯云) | 257.958 | 228.325
福建厦门(移动) | 60.429 |  
河南许昌(ptgidc.com) | 161.141(丢包:25) | 313.299(丢包:50)
四川成都(移动) | 87.051(丢包:20) | 241.622(丢包:20)
上海(电信) | 136.827(丢包:50) | 213.361
广西钦州(联通) | 103.407 | 277.163
广东茂名(电信) | 235.619(丢包:25) | 213.146(丢包:25)
北京(阿里云) | 218.799(丢包:25) | 164.126(丢包:25)
福建厦门(移动云) | 55.663(丢包:25) |  
江苏无锡(电信-云端) | 89.228(丢包:50) | 228.092
青海西宁(天翼云) | 96.669 | 195.293(丢包:50)
上海(天翼云四区) | 80.646 | 186.163
广东汕头(4090.cn) | 243.692(丢包:50) | 219.333
江西南昌(天翼云) | 81.322(丢包:25) | 204.625(丢包:75)
江苏徐州(联通) | 99.136(丢包:25) | 289.458(丢包:25)
贵州贵阳(电信) | 260.972(丢包:50) | 241.662
新疆乌鲁木齐(天翼云) | 130.687(丢包:50) | 305.842(丢包:50)
广东深圳(移动) | 20.936 | 185.518(丢包:75)
广东广州(天翼云二区) | 176.011 | 214.651(丢包:25)
安徽滁州(电信2) | 193.282(丢包:25) | 241.834(丢包:75)
江苏南京(腾讯云-轻量) | 135.991 | 231.190
重庆(电信) | 228.239(丢包:75) | 233.250
江苏徐州(移动) | 73.896(丢包:50) | 285.522(丢包:25)
江苏南通(移动) | 163.443 | 333.310(丢包:50)
辽宁沈阳LM2(移动) | 54.894 | 217.110(丢包:50)
湖南长沙(联通) | 88.979(丢包:25) | 263.326(丢包:25)
广东茂名(联通) | 121.653 | 261.311(丢包:25)
广东深圳(阿里云) | 181.193(丢包:50) | 215.136(丢包:25)
江苏常州(电信) | 141.004(丢包:50) | 226.269(丢包:25)
上海联通(yunlifang.cn) |   |  
上海(腾讯云) | 91.364(丢包:25) | 214.024(丢包:25)
四川德阳联通(yunlifang.cn) |   |  
辽宁沈阳(电信) |   |  
浙江杭州(bgp) | 132.042(丢包:25) |  
河北石家庄(天翼云) | 76.273(丢包:50) | 249.519(丢包:50)
内蒙古呼和浩特(bgp) | 204.794(丢包:25) | 199.045(丢包:50)
浙江金华(联通) | 79.591 | 295.943
河南郑州(天翼云) | 84.818(丢包:25) | 184.216(丢包:25)
辽宁大连(电信) | 90.398 | 226.829(丢包:25)
江苏无锡(电信) | 117.058 | 240.037(丢包:50)
山东青岛(阿里云) | 81.921(丢包:50) | 317.462
云南昆明(移动) | 60.108(丢包:75) | 211.945(丢包:25)
重庆(移动云) | 34.507 |  
重庆(腾讯云) | 205.781(丢包:25) | 238.509(丢包:75)
广东广州(滴滴云一区) | 219.939 | 205.897
上海(阿里云) | 106.292(丢包:25) | 345.599
江苏苏州(移动-云端) |   |  
北京(移动云) | 62.704(丢包:50) | 200.308(丢包:75)
黑龙江哈尔滨(天翼云) | 89.453(丢包:25) | 197.745
江苏宿迁(联通) | 80.073 | 287.882(丢包:25)
广东广州(天翼云四区) | 231.766(丢包:50) | 205.890(丢包:25)
广东广州(上海云盾) | 231.247(丢包:50) | 209.110(丢包:50)
江苏镇江(移动) | 43.464(丢包:50) | 235.193(丢包:25)
江苏镇江(59盾) |   |  
江苏镇江(联通) |   |  
上海(移动云) | 48.851 | 225.339
宁夏中卫(天翼云) | 78.905(丢包:50) | 217.923(丢包:25)
福建福州电信(yunlifang.cn) |   |  
天津(腾讯云) | 68.163 | 167.010(丢包:25)
云南昆明(电信) | 268.499(丢包:50) | 237.324
辽宁沈阳(联通) |   |  
内蒙古呼和浩特(蓝视云) | 196.960(丢包:25) | 244.609(丢包:25)
四川成都(移动云) | 56.054(丢包:25) |  
江苏无锡(移动云) | 43.635 | 262.681(丢包:50)
北京(金山云) | 62.867(丢包:25) | 169.879(丢包:25)
江苏镇江(电信-亿速云) | 105.854 | 228.173(丢包:75)
上海(bgp) | 37.028 | 267.004
辽宁大连(电信) | 89.254 | 176.561(丢包:25)
福建福州(天翼云) | 74.305(丢包:50) | 260.349(丢包:50)
天津(腾讯云-湘情盾) | 63.750(丢包:50) | 167.658
四川成都(腾讯云-轻量) | 209.414(丢包:75) | 220.543
江苏苏州(移动云) | 56.409(丢包:75) |  
福建泉州(电信) | 115.934 | 228.351(丢包:25)
江苏扬州(59盾) |   |  
广东广州(移动云) | 9.411(丢包:25) | 41.185(丢包:75)
江苏苏州(电信2-云端) | 153.547 | 223.452
广东深圳(电信-无忧云) | 234.336 | 213.045(丢包:50)
山东枣庄(联通) | 158.294(丢包:50) | 316.647(丢包:25)
北京(滴滴云) | 66.988(丢包:25) | 303.884(丢包:50)
四川成都(阿里云) | 178.094 | 321.741(丢包:75)
上海地面通(CN2) | 33.310 | 33.178
广西南宁(天翼云) | 248.387 | 216.645
吉林长春(铁通) | 75.690 | 230.948(丢包:50)
湖南娄底(联通) | 101.591 | 306.724(丢包:50)
上海(青云) | 107.812(丢包:75) | 300.675(丢包:25)
安徽合肥(移动) | 28.099 |  
陕西西安(天翼云二区) | 105.663 | 226.170(丢包:50)
四川德阳(电信) | 235.264(丢包:50) | 227.553
湖南益阳(移动) | 62.893(丢包:25) | 209.744(丢包:75)
贵州贵阳(移动云) | 35.527(丢包:25) | 209.446(丢包:75)
江苏镇江(电信-云朵艾欧) | 123.092 | 239.943(丢包:25)
浙江金华(移动-华晨网络) | 56.088 |  
吉林长春(天翼云) | 139.171(丢包:50) | 261.264(丢包:25)
江西南昌(电信) | 204.730(丢包:75) | 229.154(丢包:25)
北京(青云3b) | 38.770 | 39.468
广东东莞(电信CN2) |   |  
福建福州(联通) | 116.347 | 283.929(丢包:50)
广西南宁(电信) | 200.065 | 218.637(丢包:75)
辽宁沈阳(天翼云三区) | 76.148 | 221.383(丢包:50)
广东广州(滴滴云二区) | 217.125(丢包:75) | 225.657(丢包:25)
内蒙古呼和浩特(天翼云三区) | 203.143(丢包:25) | 252.274(丢包:25)
湖南常德电信(yunlifang.cn) | 38.423 | 39.097
广东深圳(天翼云) | 191.054(丢包:50) | 232.562(丢包:25)
湖南长沙(天翼云) | 201.742(丢包:25) | 233.440(丢包:25)
河南郑州(移动云) | 53.882(丢包:25) | 219.196(丢包:50)
浙江金华(电信) | 190.407 | 216.146(丢包:25)
江苏宿迁(电信) | 136.428 | 251.462
浙江宁波(移动) |   |  
辽宁沈阳(联通-云端2) | 123.934 | 371.616
云南昆明(天翼云) | 222.215(丢包:25) | 259.922
江苏苏州(天翼云) | 145.791 | 297.329(丢包:50)
山东济南(移动云) | 40.463(丢包:25) | 210.367
江苏宿迁(移动) | 51.515 |  
江苏常州(移动) | 60.041 | 272.038
广东广州(腾讯云) | 175.058(丢包:25) | 206.080(丢包:25)
河北保定(联通) | 97.561 | 381.245
江苏徐州(bgp) | 195.620 | 239.530(丢包:50)
北京(青云三区A) | 110.444(丢包:25) | 125.016
广东广州(bgp) | 18.910(丢包:25) | 205.397
江苏无锡(bgp) | 39.232(丢包:25) | 290.587(丢包:25)
浙江宁波(电信-云端) | 169.796(丢包:75) | 216.789(丢包:50)
贵州贵阳(天翼云) | 249.766(丢包:25) | 211.779
辽宁沈阳(移动云) | 48.887 | 211.709(丢包:75)
辽宁鞍山(电信) | 74.720(丢包:25) | 186.290(丢包:50)
天津(联通) | 150.264(丢包:50) | 313.561(丢包:25)
江苏宿迁(bgp) | 207.647(丢包:50) | 244.791
四川绵阳(明源网络) | 266.393(丢包:75) | 250.671(丢包:25)
云南昆明(移动云) | 36.723 | 38.864(丢包:75)
浙江杭州(移动云) | 33.043(丢包:25) |  
福建厦门(联通) | 96.813 | 276.243(丢包:25)
江苏南通(电信) | 102.073(丢包:50) | 226.846(丢包:50)
重庆(天翼云) | 210.376(丢包:75) | 238.020
湖北武汉(天翼云) | 197.083(丢包:25) | 219.097
安徽滁州(电信) | 186.930 | 240.704(丢包:75)
河南新乡(bgp-7yc) | 160.498 | 306.456(丢包:75)
陕西西安(移动云) | 55.828 | 216.455(丢包:50)
广东佛山(电信) | 178.012(丢包:50) | 216.400(丢包:25)
江苏宿迁(电信-7yc) | 95.503(丢包:25) | 232.485(丢包:75)
福建福州(电信) | 78.240 | 245.541
江苏无锡(联通) | 98.859 | 270.701(丢包:50)
广东深圳(momayun) | 227.319 | 198.165
上海(华为云) | 153.298 | 211.413(丢包:50)
山东滨州(联通-云端) | 116.863 | 268.995(丢包:50)
江苏南京(天翼云) | 104.033 | 220.949(丢包:25)
贵州贵阳(华为云) | 127.236 | 273.266
上海(腾讯云-轻量) | 116.638(丢包:50) |  
山西太原(天翼云) | 74.190(丢包:50) | 171.524(丢包:25)
安徽马鞍山(移动) | 30.823 |  
四川成都(天翼云) | 211.157(丢包:75) | 230.538(丢包:25)
江苏徐州(移动-云端) | 78.475(丢包:20) |  
辽宁沈阳(移动) | 61.326 |  

</details>

****

只可惜大半夜的，延迟波动、丢包问题无解，而且这个 IP 可能很快就动态调整了 ......

****

早上起来，这个 IP 还没有被动态调整，测速结果如下：

``` bash
开始延迟测速（模式：TCP IPv4，端口：443）：
10 / 10 [-----------------------------------------------------------------------------------------------------] 100.00%
开始下载测速（延迟时间上限：9999.00 ms，下载速度下限：0.00 MB/s）：
1 / 1 [-------------------------------------------------------------------------------------------------------] 100.00%
IP 地址           已发送  已接收  丢包率  平均延迟  下载速度 (MB/s)
172.64.228.123    10     9      0.10    103.71    4.53

# 相比其他的 IP 主要是延迟低，速度一般般（我测速最快的 IP 起码有 25MB/s），而且还伴随丢包...
# 这个 IP 段共有 256 个 IP，最后一段的 123 随便选个（0~255），我只是随便选了一个。
```

****

又测试了几次，发现即使是白天，这个 IP 段依然存在**少许丢包**（约 10%，晚上约 30%，这可能是走 NTT 垃圾线路的原因），怪不得平时测速很难见到这种日本 IP 段，估计是因为略微丢包，导致被**排序到后面**去了（延迟排序时，会根据丢包率分开排序）。

> 这告诉我一个道理，有时候一些**好东西会被不经意间藏起来**，或许翻一翻 `result.csv` 完整测速结果文件会有惊喜。

****

突然发现这个 IP 段在**其他国家访问时就变成其他地区**了（Anycast 任播技术），国内其他地区没测试过，应该都还是日本。

**Comment by CrazyBoyFeng** [comment]
呃，我开这个贴的主要目的不是讨论如何找最快IP，而是想探讨同一个IP的适用范围多广。也就是，需不需要为北京移动、天津移动、上海移动、广东移动等等的IP分别进行维护。是需要分省维护还是分大区维护。  
可能很多人拿到IP后的使用方式是自用，用不着关心这个。不过还是有用来加速网站的。因为网站的用户分布很广，所以需要考虑分区域的问题。

另外其实也不建议公开分享测到的IP，很多人用的话，马上就会变得很慢。大家自测自用，分散一点比较好。

**Reply by XIU2** [reply]
我前面也做过测试，在国内大部分地区，该 IP 段的**延迟优于**默认分配 IP，因此该 IP 段的**适用范围为全国**。 

其实这个日本的 IP 段**下载测速并不快**（垃圾 NTT 线路），只是**延迟低一些**罢了，而且还伴随**少许丢包**...
另外，这是一整个 IP 段（共 256 个 IP），而不是单独一个 IP，所以影响不大。

****

虽然我知道最好是捂着自己用，但是我这个人特别喜欢分享帮助别人，根本藏不住好东西，导致每次我遇到或者写出好东西，就想第一时间分享出去...

**Comment by denniszzg** [comment]
首先感谢各位程序员们的贡献，没有你们的努力，世界会变得多麻烦啊。单从科学上网来说，能翻墙的人数要少一个数量级。这是很让人悲哀的一件事。

从这个意义上来说，你们创造的工具越易用，普适性越强就越好。曲高和寡，它的价值就大打折扣了。

**我觉得创造一个项目，能够分地区，运营商，测出的优质ip能够自动分享上来，让没有技术能力的人也能简单地改善翻墙质量。这是利大于弊的。**
    - 参考 [BT trackerlist项目](https://ngosang.github.io/trackerslist/)
    - 这样做是否存在什么隐患，需要大佬们讨论

有的程序员存在鄙视链思维，认为不懂技术或不愿钻研技术翻什么墙，甚至特意用技术门槛过滤掉“低端”用户（很明显，XIU2同志不是这一类人 ^_^）。从逻辑上说，使翻墙工具更易于使用 -> 更多的人能翻墙 -> 善莫大焉

**Reply by XIU2** [reply]
哈~ 我没有那么高尚，我目前分享出来的软件、脚本等乱七八糟的，都是我自用的（为了自给自足特地自学了各种脚本、编程语言），只不过我这个人比较喜欢分享、帮助别人。

**Reply by CrazyBoyFeng** [reply]
这应该是评论我楼上不建议分享ip的观点。  
我的考虑是，测出的快速ip，很多人用的话，会迅速变得很慢。测出的“最快”就没有意义了，因此不宜分享。这是因为某个时刻，去往同一个ip的国际链路是固定的，而这个链路的带宽容量是固定的。比如，163网去日本平常就比去美国慢，这是因为电信从ntt手里买的带宽少，日本线路拥挤。不过电信会动态调整，经常会发现去日本绕道美国。那就是因为日本线路拥挤了。而联通的动态均衡就差一些，会出现快的很快，慢的很慢。所以大家都访问一个ip，这个ip的链路就会拥挤。  
何况，这个ip对于不同省的人来说也不一定是最快。
所以要想得到最快ip，还是得用这个工具自测，且分散使用ip。

**Reply by denniszzg** [reply]
理解。我可能更多考虑翻墙这一块儿了，毕竟，以前完全不知道还可以自己搜索IP，使用体验非常糟糕。网速快一点，获取信息就快一点，时间就是生命啊。再次致谢各位程序员。
